### ISSUE TITLE
[Feature Request] Designated initializers

### ISSUE THREAD

**Issue by billgan1024** [microsoft/DirectXShaderCompiler#5866]
**Is your feature request related to a problem? Please describe.**
Describe the problem that led you to request this feature in as much detail as you can.

Tried to use c++20 designator initializer syntax in HLSL shader model 6 but it doesn't seem to be supported. For example, 
```
vs_out vs_main(vec4 position : POSITION, vec4 color : COLOR)
{
    vs_out output = {
        .position = position,
        .color = color,
    };
    return output;
}
```
should be equivalent to 
```
vs_out vs_main(vec4 position : POSITION, vec4 color : COLOR)
{
    vs_out output;
	output.position = position;
	output.color = color;
    return output;
}
```

**Describe the solution you'd like**
C++20 designated initializers allow structs to have their fields initialized without specifying the variable name too many times,
and the remaining fields are automatically zeroed. Implement the same behaviour in HLSL.

**Comment by devshgraphicsprogramming** [comment]
The language spec repo already has an issue about this https://github.com/microsoft/hlsl-specs/issues/23

**Comment by llvm-beanz** [comment]
Duplicate of: https://github.com/microsoft/hlsl-specs/issues/23

HLSL is based on C99 with some C++98 features added in. We've got a long way to go before we can introduce C++20 features, and we'll never be supporting C++20 features in the DXC codebase, those will only be supported in Clang.
